packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -270,9 +271,18 @@ export const createWebpackUtils = (
     },
 
     js: options => {
+      let reactMajorVersion = 0
+
+      try {
+        const react = require(`react`)
+        reactMajorVersion = semver.major(react.version)
+      } catch (e) {
+        // no-op.
+      }
```

Comment:
I rather not load react the lib here as it's expensive to just get a version. Could we load in the package.json instead?
```suggestion
      try {
        const react = require(`react/package.json`)
        reactMajorVersion = semver.major(react.version)
      } catch (e) {
        // no-op.
      }
```